services/QuillCMS/app/workers/refresh_responses_view_worker.rb

```diff
@@ -1,15 +1,21 @@
-class RefreshGradedResponsesWorker
+class RefreshResponsesViewWorker
   include Sidekiq::Worker
   sidekiq_options queue: SidekiqQueue::LOW
 
   REFRESH_TIMEOUT = ENV["VIEW_STATEMENT_TIMEOUT"] || '10min'
+  VIEWS = %w(GradedResponse MultipleChoiceResponse)
+
+  class InvalidView < StandardError; end
 
   # Make the DB statement timeout longer while refreshing the materialized view
-  def perform
+  def perform(view)
     original_timeout = db_timeout
```

Comment:
To be honest, I went back and forth on where to keep the wrapper. I kept it in the job since we don't need the wrapper for `test` and `development` environments, it's only needed on `production` and `staging`.  The test suite has to call `refresh` relatively often, so I thought this made tests slightly less complicated. I could be convinced otherwise.

Also, I saw the `establish_connection` method, but that's for every call by the model, even queries, so I'd rather keep this scoped to refresh.

Reply:
Those are good reasons to keep it where you have it, especially the point about queries.